Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,431 @@
+            result.setEndPoints(start, end);
+            return result;
+        }
+        return new EllipsoidRhumbLine(start, end, e);
+    };
+
+    /**
+     * Create a rhumb line using an initial position with a heading and distance.
+     *
+     * @param {Cartographic} start The initial planetodetic point on the path.
+     * @param {Number} heading The heading in radians.
+     * @param {Number} distance The rhumb line distance between the start and end point.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the rhumb line lies.
+     * @param {EllipsoidRhumbLine} [result] The object in which to store the result.
+     * @returns {EllipsoidRhumbLine} The EllipsoidRhumbLine object.
+     */
+    EllipsoidRhumbLine.fromStartHeadingDistance = function(start, heading, distance, ellipsoid, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('start', start);
+        Check.defined('heading', heading);
+        Check.defined('distance', distance);
+        //>>includeEnd('debug');
+
+        var e = defaultValue(ellipsoid, Ellipsoid.WGS84);
+
+        if (defined(result)) {
+            result._ellipsoid = e;
+            initiailize(result, start, undefined, e);
+            result._heading = CesiumMath.negativePiToPi(heading);
+            result._distance = distance;
+
+            result._end = result.interpolateUsingSurfaceDistance(distance, new Cartographic());
+            return result;
+        }
+
+        var rhumbLine = new EllipsoidRhumbLine(undefined, undefined, e);
+        initiailize(rhumbLine, start, undefined, e);
+        rhumbLine._heading = CesiumMath.negativePiToPi(heading);
+        rhumbLine._distance = distance;
+        rhumbLine._end = rhumbLine.interpolateUsingSurfaceDistance(distance, new Cartographic());
+
+        return rhumbLine;
+    };
+
+    /**
+     * Sets the start and end points of the rhumb line.
+     *
+     * @param {Cartographic} start The initial planetodetic point on the path.
+     * @param {Cartographic} end The final planetodetic point on the path.
+     */
+    EllipsoidRhumbLine.prototype.setEndPoints = function(start, end) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('start', start);
+        Check.defined('end', end);
+        //>>includeEnd('debug');
+
+        computeProperties(this, start, end, this._ellipsoid);
+    };
+
+    /**
+     * Provides the location of a point at the indicated portion along the rhumb line.
+     *
+     * @param {Number} fraction The portion of the distance between the initial and final points.
+     * @param {Cartographic} result The object in which to store the result.
+     * @returns {Cartographic} The location of the point along the rhumb line.
+     */
+    EllipsoidRhumbLine.prototype.interpolateUsingFraction = function(fraction, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number('distance', this._distance);
```

Comment:
This looks like a duplicate `Check` since it's also handled below in the callee.